controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -129,6 +129,12 @@ func (ew *EndpointsWatcher) Subscribe(authority string, listener EndpointUpdateL
 	if err != nil {
 		return err
 	}
+
+	svc, _ := ew.k8sAPI.Svc().Lister().Services(id.Namespace).Get(id.Name)
```

Comment:
This same call is repeated in the same code path when not dealing with external services, in `newPortPublisher()`. From my understanding there's a shared informer underneath that caches that, but given the informer is instantiated again that caching is not happening, right? If so, you could consider shuffling things a bit to have that call happen only once.

Reply:
I don't think the informer is being instantiated again.  there should only be one informer instance, I believe

Reply:
My bad, the informer is indeed kept in `API.svc` and it's reused.